Game.java

```diff
@@ -1,53 +1,98 @@
 import java.io.InputStream;
+import java.util.Random;
 import java.util.Scanner;
 
 public class Game {
 
+    private final Scanner scanner;
     private char turn = 'o';
     private InputStream stream;
+    private Board board;
+    private String gameMode = "pvp";
 
-    public Game(InputStream stream) {
+    public Game(InputStream stream, Board board) {
         this.stream = stream;
+        this.scanner = new Scanner(stream);
+        this.board = board;
     }
 
-    public void showBoard(Board board) {
-        System.out.println("|" + board.getState().substring(0, 3) + "|");
-        System.out.println("|" + board.getState().substring(3, 6) + "|");
-        System.out.println("|" + board.getState().substring(6, 9) + "|");
+    public char getTurn() {
+        return turn;
     }
 
-    public Boolean avaliablePos(Board board, int pos)  {
-        StringBuilder currentBoard = new StringBuilder(board.getState());
-        if (currentBoard.charAt(pos - 1) == '-') {
-            if (!(pos > 9) || !(pos < 1)) {
-                System.out.print("Must be a number from 1-9 \n");
-                return true;
-            }
-        }
-        return false;
+    public void switchTurn() {
+        turn = (turn == 'o' ? 'x' : 'o');
+    }
+
+    public void print(String text) {
+        System.out.println(text);
     }
 
-    public Board playerMakesMove(Board board, int pos) {
-        if (!avaliablePos(board, pos)) return board;
-        System.out.println("Its your turn " + turn + "!");
-        turn = (turn == 'x' ? 'o' : 'x');
-        board = board.playerMakesMove(turn, pos - 1);
+    public String showBoard(Board board) {
+        print("|" + board.getState().substring(0, 3) + "|\n|" + board.getState().substring(3, 6) + "|\n|" + board.getState().substring(6, 9) + "|");
+        return String.format("|" + board.getState().substring(0, 3) + "|\n|" + board.getState().substring(3, 6) + "|\n|" + board.getState().substring(6, 9) + "|");
+    }
+
+    public Board playerMakesMove() {
+        int index = randInt(0, board.availablePositions().size());
+        int pos;
+        if (turn == 'o' && gameMode.equals("pvc")) {
```

Comment:
you use _turn_ here and not _getTurn()_, but on line 48 you use _getTurn()_, be consistent.